scripts/pre_commit_linter.py

```diff
@@ -1478,6 +1511,88 @@ def _check_directive_scope(all_files, parsed_js_files):
     return summary_messages
 
 
+def _check_js_component_count(all_files, parsed_js_files):
+    """This function ensures that all JS files have exactly
+    one component and and that the name of the component
+    matches the filename.
+    """
+    print 'Starting js component count check'
+    print '----------------------------------------'
+    # Select JS files which need to be checked.
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js') and not filename.endswith('App.js')]
+    failed = False
+    summary_messages = []
+    component_name = ''
+    components_to_check = ['controller', 'directive', 'factory', 'filter']
+    for filename in files_to_check:
+        property_num = 0
+        # Filename without its path.
+        exact_filename = filename.split('/')[-1][:-3]
+        parsed_dict = parsed_js_files[filename]
+        with _redirect_stdout(_TARGET_STDOUT):
+            # Parse the body of the content as nodes.
+            parsed_nodes = parsed_dict['body']
+            for parsed_node in parsed_nodes:
+                expression = _get_expression_from_node_if_one_exists(
+                    parsed_node, components_to_check)
+                if not expression:
+                    continue
+                # Separate the arguments of the expression.
+                arguments = expression['arguments']
+                # The first argument of the expression is the
+                # name of the component.
+                component_name = arguments[0]['value']
+                property_num += 1
+                component = expression['callee']['property']['name']
+                if component == 'directive':
+                    if (component_name[0].upper() + component_name[1:] +
+                            'Directive' != (exact_filename)):
+                        print (
+                            '%s -> Please ensure that the %s name strictly '
+                            'matches the filename'
+                            % (filename, component))
+                        failed = True
+                elif component == 'filter':
```

Comment:
I see a repetition of code here. You might want to club these two?

Reply:
Done